### PR TITLE
add BrowserUtils.isIE() method

### PR DESCRIPTION
**Pull request by mrjones2014**
- [x] Related GitHub issue(s) linked in PR description
- [x] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [x] Code builds cleanly (no _additional_ warnings or errors)
- [x] Manually tested
- [x] Automated tests are passing
- [x] No _decreases_ in automated test coverage
- [x] Documentation updated (readme, docs, comments, etc.)
- [x] Localization: No hard-coded error messages in code files (_minimally_ in string constants)

fixes: #23

### CODE REVIEW

File: src/utilities/browser-utils.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+}
+/**
+* Returns true if the detected browser is Internet Explorer.
+* @returns boolean
+*/
+const _isIE = (): boolean => {
+  return /*@cc_on!@*/ false || !!document.documentMode;
+};
+
+const _isNotIE = (): boolean => !_isIE();
```
Comment (wintondeshong) [inline src/utilities/browser-utils.ts:17]: Please provide comment for function

File: src/utilities/browser-utils.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/**
+ * Allows us to use the documentMode for feature detection on globalThis.Document
+ */
+declare global {
+  interface Document {
+      documentMode?: any;
+  }
+}
```
Comment (wintondeshong) [inline src/utilities/browser-utils.ts:8]: Simple newline for readability after line 8

File: src/utilities/browser-utils.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  interface Document {
+      documentMode?: any;
+  }
+}
+/**
+* Returns true if the detected browser is Internet Explorer.
+* @returns boolean
+*/
+const _isIE = (): boolean => {
+  return /*@cc_on!@*/ false || !!document.documentMode;
```
Comment (wintondeshong) [inline src/utilities/browser-utils.ts:14]: While you won't be able to get branch coverage for `/*@cc_on!@*/`, you could mock document.documentMode to test this function. If time permits, please write some unit tests for these methods.

File: src/utilities/browser-utils.ts (hunk 2 above)
Comment (mrjones2014) [reply src/utilities/browser-utils.ts:8]: Fixed.

File: src/utilities/browser-utils.ts (hunk 3 above)
Comment (mrjones2014) [reply src/utilities/browser-utils.ts:15]: Added tests.

File: src/utilities/browser-utils.test.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      // Act
+      const result = BrowserUtils.isNotIE();
+
+      // Assert
+      expect(result).toBeFalse();
+    });
+  });
+
+  // Clean up
+  window.document.documentMode = undefined;
```
Comment (wintondeshong) [inline src/utilities/browser-utils.test.ts:53]: Do you want this after each test? If so use 'afterEach` and put at top of current describe block.

**Comment by wintondeshong** [comment]
@mrjones2014 let me know when you are good with most recent feedback and I'll merge and publish.

File: src/utilities/browser-utils.test.ts (hunk 4 above)
Comment (mrjones2014) [reply src/utilities/browser-utils.test.ts:53]: @wintondeshong each test is setting the value to whatever it needs to be in it's `// Arrange` section.

Comment (mrjones2014) [reply src/utilities/browser-utils.test.ts:53]: @wintondeshong each test is setting the value to whatever it needs to be in it's `// Arrange` section.

Comment (wintondeshong) [reply src/utilities/browser-utils.test.ts:53]: @mrjones2014 yeah, then you'll need to call `` in an afterEach in the top-level describe. I'll merge and make the change

```javascript
afterEach(() => {
    window.document.documentMode = undefined;
});
```

**Comment by wintondeshong** [comment]
Published with version 0.1.3

**Comment by brandongregoryscott** [comment]
@all-contributors add @mrjones2014 for code, test

**Pull request merged**